src/warc2zim/utils.py

```diff
@@ -42,14 +53,43 @@ def parse_title(content):
         return ""
 
 
-def to_string(input_: str | bytes) -> str:
-    try:
-        input_ = input_.decode(  # pyright: ignore[reportGeneralTypeIssues, reportAttributeAccessIssue]
-            "utf-8-sig"
-        )
-    except AttributeError:
-        pass
-    return input_  # pyright: ignore[reportGeneralTypeIssues, reportReturnType]
+def get_record_encoding(record: ArcWarcRecord) -> str | None:
+    content_type = get_record_content_type(record)
+    if m := ENCODING_RE.search(content_type):
```

Comment:
This will never work. that regexp looks for `encoding=xxx` pattern while your content-type is _just_ the value from the headers

Reply:
Regex searches for "charset=xxx" or "encoding=xxx" and if content-type is indeed just the value from the headers, this is the way to declare encoding in the header:
https://www.w3.org/International/articles/http-charset/index

Reply:
My bad 👍